ante/fee.go

```diff
@@ -0,0 +1,82 @@
+package ante
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	oracletypes "github.com/umee-network/umee/x/oracle/types"
+)
+
+// MaxOracleMsgGasUsage is constant expected oracle msg gas cost
+const MaxOracleMsgGasUsage = uint64(100_000)
+
+// MempoolFeeDecorator will check if the transaction's fee is at least as large
+// as the local validator's minimum gasFee (defined in validator config).
+// If fee is too low, decorator returns error and tx is rejected from mempool.
+// Note this only applies when ctx.CheckTx = true
+// If fee is high enough or not CheckTx, then call next AnteHandler
+// CONTRACT: Tx must implement FeeTx to use MempoolFeeDecorator
+type MempoolFeeDecorator struct{}
+
+func NewMempoolFeeDecorator() MempoolFeeDecorator {
+	return MempoolFeeDecorator{}
+}
+
+func (mfd MempoolFeeDecorator) AnteHandle(
+	ctx sdk.Context,
+	tx sdk.Tx,
+	simulate bool,
+	next sdk.AnteHandler,
+) (newCtx sdk.Context, err error) {
+	feeTx, ok := tx.(sdk.FeeTx)
+	if !ok {
+		return ctx, sdkerrors.Wrap(sdkerrors.ErrTxDecode, "Tx must be a FeeTx")
+	}
+
+	feeCoins := feeTx.GetFee()
+	gas := feeTx.GetGas()
+	msgs := feeTx.GetMsgs()
+	// Ensure that the provided fees meet a minimum threshold for the validator,
+	// if this is a CheckTx. This is only for local mempool purposes, and thus
+	// is only ran on check tx.
+	if ctx.IsCheckTx() && !simulate &&
+		!(isOracleTx(msgs) && gas <= uint64(len(msgs))*MaxOracleMsgGasUsage) {
```

Comment:
`!(isOracleTx(msgs)`
Is this check for non-oracle transactions?

Reply:
Yes, so we only apply the minimum fee check if the tx contains non-oracle messages. In other words, oracle txs get a free pass.

Reply:
It looks like if a tx contained 1 oracle message and also some non-oracle messages, it would also get a pass.

How should such "mixed messages" be treated? (or is it an issue at all?)